lib/vapor/config/env.ex

```diff
@@ -1,11 +1,60 @@
 defmodule Vapor.Config.Env do
+  @moduledoc """
+  The Env config module provides support for pulling configuration values
+  from the environment. It can do this by either specifying a prefix or by
+  specifying specific bindings from keys to environment variables.
+
+  ## Loading configuration by prefix
+
+  If a prefix is used then the variable will be normalized by removing the
+  prefix, downcasing the text, and converting all underscores into nested keys
+
+  ## Loading with bindings
+
+  Specific bindings be also be specified. They are case sensitive
+  """
+
   defstruct prefix: :none, bindings: []
 
+  @type bindings :: Keyword.t(String.t)
+
+  @type t :: %__MODULE__{
+    prefix: :none | String.t,
+    bindings: bindings,
+  }
+
+  @spec with_prefix(String.t) :: t
   def with_prefix(prefix) when is_binary(prefix) do
-    %__MODULE__{prefix: "#{prefix}_"}
+    %__MODULE__{prefix: build_prefix(prefix)}
+  end
+
+  @doc """
+  Creates a configuration plan with explicit bindings.
+
+  Env.with_bindings([foo: "FOO", bar: "BAR"])
+  """
+  @spec with_bindings(bindings()) :: t
+  def with_bindings(opts) when is_list(opts) do
+    %__MODULE__{
+      prefix: :none,
+      bindings: opts,
+    }
   end
 
+  defp build_prefix(prefix), do: "#{prefix}_"
+
   defimpl Vapor.Provider do
+    def load(%{prefix: :none, bindings: bindings}) do
```

Comment:
I want to look to see _when_ `load` happens in the lifecycle. My assumption is that it happens at application start time, but I'm not 100% sure. _If_ it is only at application start time, then it might be worth adding a test where we explicitly change the value of an environment variable _after_ `load` and ensure the value that comes from `Config` is the _original_.

Reply:
Currently `load` is called when the users vapor config process is started. Typically this will probably be early in the boot process of the users application. But its up to them.

I agree with your point about having a test that shows that we aren't pulling from the environment each time. I'll add that to the criteria in #5.